test/unit/component.js

```diff
@@ -82,142 +85,374 @@ describe('Component class', () => {
+            assert.deepEqual(component.googleFonts, [...googleFonts1, ...googleFonts2]);
+          });
+        });
+
+        describe('viewData', () => {
+          it('returns merged object of model, viewData, classes, and text', () => {
+            component.model = {model: 'model'};
+            component = Object.defineProperty(component, 'classes', {
+              value: 'classes',
+            });
+            const expectedObj = {
+              viewData: component.options.viewData.viewData,
+              text: component.options.text,
+              model: component.model.model,
+              classes: component.classes,
+            };
+            assert.deepEqual(component.viewData, expectedObj);
+          });
         });
       });
-    });
 
-    describe('updateConfig()', () => {
-      it('delegates to updater', () => {
-        const component = new Component({id: 1234});
-        component.updater.updateConfig = sinon.spy();
-        component.updateConfig({test: 'lol'});
-        assert.calledWith(component.updater.updateConfig, {test: 'lol'});
+      describe('morphCallbacks', () => {
+        it('returns an object with the function onBeforeElUpdated', () => {
+          assert.instanceOf(component.morphCallbacks, Object);
+          assert.equal(Object.keys(component.morphCallbacks).length, 1);
+          assert.instanceOf(component.morphCallbacks.onBeforeElUpdated, Function);
+        });
+
+        describe('onBeforeElUpdated()', () => {
+          it('returns false if fromEl\'s tagname is img and its source is toEl\'s data-src element', () => {
+            const fromEl = {tagName: 'IMG', src: 'data-src'};
+            const toEl = {
+              getAttribute(param) {
+                return param;
+              },
+            };
+            assert.equal(component.morphCallbacks.onBeforeElUpdated(fromEl, toEl), false);
+          });
+
+          it('returns true if fromEl\'s tagname is not img or its source is not toEl\'s data-src element', () => {
+            const fromEl = {tagName: 'not IMG', src: 'not data-src'};
+            const toEl = {
+              getAttribute(param) {
+                return param;
+              },
+            };
+            assert.equal(component.morphCallbacks.onBeforeElUpdated(fromEl, toEl), true);
+          });
+        });
       });
     });
 
-    describe('destroy()', () => {
-      it('delegates to view', () => {
-        const component = new Component({id: 1234});
-        component.view.destroy = sinon.spy();
-        component.destroy();
-        assert.calledOnce(component.view.destroy);
+    describe('"private" methods', () => {
+      describe('_userEvent()', () => {
+        it('logs to logger if debug is set to true', () => {
+          const infoSpy = sinon.spy();
+          const loggerStub = sinon.stub(logger, 'default').value({info: infoSpy});
+          component.globalConfig.debug = true;
+          component.typeKey = 'key';
```

Comment:
Do we need to change the typekey here, or could we just use the once set on the prototype?

Reply:
I added it because the text uses the typeKey so I wanted to be more explicit about it